### DISCUSSION TITLE
How can I emit a new event from a Subscriber impl?

### DISCUSSION THREAD

**Discussion by andrewbanchich** [tokio-rs/tracing#2741] [Q&A]
I would like to create a subscriber which either mutates events or ignores them and logs new ones.

Is the only / best way of doing this to spawn a new Tokio task and emit the event from there?

**Comment by kaffarell** [comment]
Just use the `event!()` macro to emit a new event. Mutating events isn't possible as far as I know.

**Reply by andrewbanchich** [reply]
> Just use the event!() macro to emit a new event

that doesn't work

**Reply by kaffarell** [reply]
Why would `tracing::event!(..)` not work?

**Reply by andrewbanchich** [reply]
I'm not a core tracing dev, so I'm not sure - but it doesn't emit any event if called in a Subscriber impl.

**Reply by kaffarell** [reply]
Then there is probably something wrong with your Subscriber and/or Filter. I just tried it and it worked with a minimal example. Maybe you could post a minimal snippet of your code so I can have a look at it?

**Reply by andrewbanchich** [reply]
Emitting an event from the `event()` handler in `Subscriber` impls causes [a stack overflow](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=99ee62300571fe462a3c8c09e10b6c80). Using `println!` does not have this issue since it doesn't cause recursion.

**Reply by kaffarell** [reply]
Hmm, yeah well you can't prevent this...
